EventFilter/L1TRawToDigi/plugins/implementations_stage2/EMTFSetup.cc

```diff
@@ -68,11 +68,13 @@ namespace l1t {
 
          auto emtf_headers_unp   = UnpackerFactory::get()->make("stage2::emtf::HeadersBlockUnpacker");  // Unpack "AMC data header" and "Event Record Header"
          auto emtf_counters_unp  = UnpackerFactory::get()->make("stage2::emtf::CountersBlockUnpacker"); // Unpack "Block of Counters"
-         auto emtf_me_unp       = UnpackerFactory::get()->make("stage2::emtf::MEBlockUnpacker");      // Unpack "ME Data Record"
-         auto emtf_rpc_unp      = UnpackerFactory::get()->make("stage2::emtf::RPCBlockUnpacker");     // // Unpack "RPC Data Record"
-         auto emtf_sp_unp       = UnpackerFactory::get()->make("stage2::emtf::SPBlockUnpacker");      // Unpack "SP Output Data Record"
+         auto emtf_me_unp       = UnpackerFactory::get()->make("stage2::emtf::MEBlockUnpacker");        // Unpack "ME Data Record"
```

Comment:
If you cast this pointer to a MEBlockUnpacker
`emtf_me_unp = static_pointer_cast<stage2::emtf::MEBlockUnpacker>(UnpackerFactory::get()->make("stage2::emtf::MEBlockUnpacker"));`
you can move the code changes from Unpacker.h to EMTFBlockME.cc so that only the EMTF unpacker code is changed.

Reply:
I decided to put the changes in Unpacker.h so that all unpackers have the ability to use this rather general functionality.  It could easily be used for other EMTF blocks, or other subsystems in the future.  It certainly doesn't hurt or add overhead to the other unpackers, right?

Reply:
Yes if there are plans that this could be used in other unpackers as well. But it adds a variable that most of them do not use.
If we keep it here the setters and getters can be removed from MuonUnpacker.h.

Reply:
Well, that's at least 2 systems - EMTF and uGMT :)
To me, it's cleaner to have the function only in Unpacker.h, rather than having duplicates in MuonUnpacker.h, IntermediateMuonUnpacker.h, and some (not yet existing) EMTFUnpacker.h file.